src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -47,42 +41,30 @@ public WorkItem(SendOrPostCallback callback, object? data, ManualResetEventSlim?
             }
         }
 
-        internal JSSynchronizationContext(Thread targetThread, IntPtr targetThreadId)
-            : this(
-                targetThread, targetThreadId,
-                Channel.CreateUnbounded<WorkItem>(
-                    new UnboundedChannelOptions { SingleWriter = false, SingleReader = true, AllowSynchronousContinuations = true }
-                )
-            )
-        {
-        }
-
-        internal static void AssertWebWorkerContext()
-        {
-#if FEATURE_WASM_THREADS
-            if (CurrentJSSynchronizationContext == null)
-            {
-                throw new InvalidOperationException("Please use dedicated worker for working with JavaScript interop. See https://aka.ms/dotnet-JS-interop-threads");
-            }
-#endif
-        }
-
-        private JSSynchronizationContext(Thread targetThread, IntPtr targetTID, WorkItemQueueType queue)
+        internal JSSynchronizationContext(bool isMainThread)
         {
-            TargetThread = targetThread;
-            TargetTID = targetTID;
-            Queue = queue;
+            ProxyContext = new JSProxyContext(isMainThread, this);
+            Queue = Channel.CreateUnbounded<WorkItem>(new UnboundedChannelOptions { SingleWriter = false, SingleReader = true, AllowSynchronousContinuations = true });
             _DataIsAvailable = DataIsAvailable;
+
+            previousSynchronizationContext = Current;
+            SetSynchronizationContext(this);
```

Comment:
(you don't have to change it in this pr) just archiving for posterity that 'the constructor registers itself as the sync context' is a potential footgun